### DISCUSSION TITLE
It seems that Android WebView actually does implement `safe-area-inset-*` environment variables

### DISCUSSION THREAD

**Discussion by vilicvane** [ionic-team/capacitor#6456] [Ideas]
I was trying to fine tune my app for full screen, and found that if you do not call the default behavior in `onApplyWindowInsetsListener`, WebView actually provides the correct `env(safe-area-inset-top)`.

However, I am not familiar with Android layout. I tried for some time but failed to make it work for navigation bar.

If you just need `env(safe-area-inset-top)`, simply do:

```ts
view.setOnApplyWindowInsetsListener((_view, insets) -> insets);
```

It might have something to do with navigation bar insets not being "stable" insets.

**Comment by nijakobius** [comment]
Where exactly do you put that code?

**Reply by vilicvane** [reply]
`MainActivity#onCreate()`, and `View view = getWindow().getDecorView();` btw.

**Reply by nijakobius** [reply]
Sry I'm not too familiar with Java - my MainActivity looks like this:

```
package com.appname.appname;

import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {}
```

Where/how exactly do I add the code? (I tried a lot and there's always some error :D)

**Reply by vilicvane** [reply]
You need to override the `onCreate()` inside `{}`:
![image](https://user-images.githubusercontent.com/970430/230776264-6b93ca8d-baab-4888-98ba-601a1bfac2bd.png)

**Reply by nijakobius** [reply]
So I got the folllowing code and the app builds but it crashes with "appname keeps stopping":

```
package com.appname.appname;

import android.os.Bundle;
import android.view.View;
import com.getcapacitor.BridgeActivity;

public class MainActivity extends BridgeActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        View view = getWindow().getDecorView();
        view.setOnApplyWindowInsetsListener((_view, insets) -> insets);
    }
}
```

Any ideas? :)

**Reply by vilicvane** [reply]
I am not Android developer either. You probably need to check out the log to see what's causing the crash.
